### PR TITLE
Home features

### PR DESCRIPTION
**Pull request by afcastaneda223**
# Documentation

All the documentation is on the README file.

## Updates:

- one page with a list of cocktails filtered by ingredients.
- one page for the item details, that include all the cocktails that the main ingredient was the filtered one.
- one page with the details of the selected cocktail

## Notes:

Coded by @afcastaneda223

✅  - No linters errors were detected.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
-[Live Version]()
+[Live Version](https://cocktailsapiapp.netlify.app/)
 
 ## Video Presentation
 
-
-[Presentation]()
+[Presentation](https://www.loom.com/share/945b95a2d3f4477ab27ff97c4eb5d676)
 
 ## Built With
```
Comment (MahmoudBakr23) [inline README.md:17]: - [ ] _Right here in this file but not the line._
1. Please provide a valid LinkedIn link in order to maintain a professional repo 😄 🙏 
2. Please give appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the author of the [original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

File: src/components/Listhome.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { getIngredients, pickIngredient } from '../redux/home/home';
+
+function Listhome() {
+  const dispatch = useDispatch();
+  const fetchApi = 'https://www.thecocktaildb.com/api/json/v1/1/list.php?i=list';
```
Comment (MahmoudBakr23) [inline src/components/Listhome.js:9]: - [ ] Your app is working great 👏 But it seems that you forgot one main feature which is the stateless filtering component that it's supposed to filter the fetched data from the API using some parameters like the category name for example. 😄 🙏 Kindly let's work on that!
_If you need a reminder please see the project requirements [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=When%20the%20page%20loads%2C%20the%20webapp%20shows%20the%20list%20of%20categories%20that%20could%20be%20filtered%20out%20by%20some%20parameter%2C%20for%20example%20by%20the%20category%20name.)._

File: src/components/__tests__/snapshot.test.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    const header = TestRenderer.create(
+      <Provider store={store}>
+        <Router>
+          <Navbar />
+        </Router>
+      </Provider>,
+    ).toJSON();
+    expect(header).toMatchSnapshot();
+  });
+});
```
Comment (MahmoudBakr23) [inline src/components/__tests__/snapshot.test.js:26]: - [ ] Well done with testing the rendering of the components so far.👏 However, you still have to write some unit and integration tests for your API functions and the Redux store by mocking. 😄 🙏 
_Please take a look at the requirements [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=Create%20unit%20tests%20for%20pure%20functions%20in%20your%20Redux%20code.)._

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @afcastaneda223 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Linter is correct 💯 
- Descriptive PR headline, summary, and commit messages 💯 
- Live demo link 💯 
- Great use of ES6 features and Redux logic 💯 
- Design followed 💯
- Great presentation (To the point, highlighted a nice piece of code) 💯 

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (afcastaneda223) [reply README.md:17]: thank you! done

File: src/components/Listhome.js (hunk 2 above)
Comment (afcastaneda223) [reply src/components/Listhome.js:9]: add state filter for ingredients in home page. also add stateless filter for language in cocktails page

File: src/components/__tests__/snapshot.test.js (hunk 3 above)
Comment (afcastaneda223) [reply src/components/__tests__/snapshot.test.js:26]: Add test mocking store and reducer

**Review by oloomoses — APPROVED** [review]
Hi @afcastaneda223, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**